ios/MusicControlManager.m

```diff
@@ -9,21 +9,56 @@
 @interface MusicControlManager ()
 
 @property (nonatomic, copy) NSString *artworkUrl;
+@property (nonatomic, strong) NSDictionary *mediaKeys;
 
 @end
 
+NSString * const MEDIA_PLAYBACK_RATE = @"playbackRate";
+
 @implementation MusicControlManager
 
 @synthesize bridge = _bridge;
 
 RCT_EXPORT_MODULE()
 
+- (NSDictionary *) mediaKeys {
+    if (_mediaKeys == nil) {
+        _mediaKeys = @{
+                @"album": MPMediaItemPropertyAlbumTitle,
+                @"trackCount": MPMediaItemPropertyAlbumTrackCount,
+                @"trackNumber": MPMediaItemPropertyAlbumTrackNumber,
+                @"artist": MPMediaItemPropertyArtist,
+                @"composer": MPMediaItemPropertyComposer,
+                @"discCount": MPMediaItemPropertyDiscCount,
+                @"discNumber": MPMediaItemPropertyDiscNumber,
+                @"genre": MPMediaItemPropertyGenre,
+                @"persistentID": MPMediaItemPropertyPersistentID,
+                @"duration": MPMediaItemPropertyPlaybackDuration,
+                @"title": MPMediaItemPropertyTitle,
+                @"elapsedPlaybackTime": MPNowPlayingInfoPropertyElapsedPlaybackTime,
+                @"playbackRate": MPNowPlayingInfoPropertyPlaybackRate,
```

Comment:
Yep, looking at the docs I got that it was getting the position from when it was last set. iOS doesn't have that property, which we can achieve by setting it in out own data structure. Which would enable us to calculate it the way you are on Android when it is not provided.

I do think it will be closer to the real thing and more consistent if the current time from the player is sent though in all situations - making it the same as the real class we making a facade for though this code.

I'll implement the states in the next hour and let you know when it's done.